test/constrained-generics/contextManager/infer-return-type.chpl

```diff
@@ -0,0 +1,16 @@
+record R : contextManager {
+  proc enterContext() ref {
+    writeln("entering");
+    return none;
```

Comment:
How is this supposed to work, returning a literal by `ref` ?  To me this should be a compilation error.

Reply:
It's not an error because of: https://github.com/chapel-lang/chapel/issues/23159

I'll see if I can generate the "wrappers" for void-typed functions, though.

Reply:
Turns out, return-type inference + any-return-intent means I don't need `none` at all, and `void` works natively :)